src/plugins/dataflow-tool/model/dataflow-program-model.ts

```diff
@@ -0,0 +1,134 @@
+import { types, Instance } from "mobx-state-tree";
+
+const ConnectionModel = types
+  .model("Connection", {
+    node: types.number,
+    output: types.maybe(types.string),
+    input: types.maybe(types.string),
+    data: types.map(types.string)
+  });
+  
+const SocketModel = types
+  .model("Socket", {
+    connections: types.map(ConnectionModel)
+  })
+  .preProcessSnapshot((snapshot: any) => {
+    // Connections are stored in rete as an array, but MST works better with maps
+    if (Array.isArray(snapshot.connections)) {
+      const connections: any = {};
+      snapshot.connections.forEach((connection: any) => {
+        connections[`${connection.node}-${connection.output}-${connection.input}`] = connection;
+      });
+      return { connections };
+    }
+    return snapshot;
+  })
+  .postProcessSnapshot((snapshot: any) => {
+    if (!Array.isArray(snapshot.connections)) {
+      return { connections: Object.values(snapshot.connections) };
+    }
+    return snapshot;
```

Comment:
```suggestion
  .postProcessSnapshot(snapshot => {
    return { connections: Object.values(snapshot.connections) };
```
I'm pretty sure the post-processor is always called with a valid MST model instance, so there should be no ambiguity about the form of its input.